### ISSUE TITLE
Homey storage should not get too full for Homey to operate (was: App settings won't store)

### ISSUE THREAD

**Issue by nlrb** [athombv/homey#1714]
**What is your Homey version (Settings → System)?**

1.5.0-rc2

**What did you try to achieve?**

*Please include detailed steps how we can reproduce the issue*
Updated protocol selection of the Weather Sensor app (enabled another protocol), which calls `Homey.manager('settings').set('protocols', setting);`.

**What was the result?**

Function returned error `Failed to save app settings! (write)`

**What did you expect as result?**

App settings to be stored like before FW 1.5.

**Any other remarks**

Don't know how to fix it.

**Please attach a Homey System log (Settings -> System -> Send Log)**

0986C15251

**Comment by nlrb** [comment]
Solved by radical intervention of @jeroenvollenbrock.

**Comment by OpenMindNL** [comment]
Could it be that this one is back with 1.5.0 rc6? 

I'm working on an update for LEDitor and I noticed that my settings are note stored anymore in the past 2 days. No error for my code, nothing in Chrome Console.... but I was testing my app with '--install'. Yesterday I installed the app with '--run' and the message '**Failed to save app settings! (write)**' started popping up. I've been checking my code for hours but can't find a clue... In an act of despair I typed the message in Google and found this issue as one and only hit.

**Comment by nattlip** [comment]
the problem of the saving of settings , of what i heard from the forum was that @nlrb had a lack of memory because of his huge logging of insights from sensors: There was no memory left to save the setings on.

**Comment by nlrb** [comment]
@OpenMindNL In my case it was indeed storage that was getting full. Homey does not handle that well. I'll re-open the case.

**Comment by nattlip** [comment]
by deleting the battery value from insights does not gives it memory free, the memory usage of a deleted value in insight doesnt change after deleting. i can see that now on the setting insights page

**Comment by OpenMindNL** [comment]
The strange thing is that when I do some edits for an animation in LEDitor 0.3.0 (the coming update), the 'Failed to save app settings! (write)' message appears at Command Prompt but the edit seems to be stored.I can open another animation and then open the editied animation again... Te edits I just did are still available. They are loaded from Homey at selection, so it seems like my edits ARE written to storage.
But... Command Prompt: Ctrl-C (stop the app)

**Comment by OpenMindNL** [comment]
The strange thing is that when I do some edits for an animation in LEDitor 0.3.0 (the coming update), the 'Failed to save app settings! (write)' message appears at Command Prompt but the edit seems to be stored.I can open another animation and then open the editied animation again... Te edits I just did are still available. They are loaded from Homey at selection, so it seems like my edits ARE written to storage.
But... Command Prompt: Ctrl-C (stop the app)

**Comment by OpenMindNL** [comment]
The strange thing is that when I do some edits for an animation in LEDitor 0.3.0 (the coming update), the 'Failed to save app settings! (write)' message appears at Command Prompt but the edit seems to be stored. I can open another animation and then open the edited animation again... The edits I just did are still available. They are loaded from Homey at selection, so it seems like my edits ARE written to storage.
But... Command Prompt: Ctrl-C (stop the app) followed directly by '--run' starts the app and shows the pre-edit animation... like no edits were done. This could be a cache-effect, so now I do the edit again in Chrome... animation should be stored... command prompt says Failed to save again... then I open Homey Desktop (Chrome still open also), go to LEDitor, open the animation... and there it is!! The edit I just did in Chrome appears in Homey Desktop?! This cannot be a cache-effect. Ok... I close Chrome and Homey Desktop... Ctrl-C at command prompt... --run .... open Chrome / Homey / LEDitor again.... edits are gone!!!.... I start running and screaming. Then I decide to remove LEDitor 0.3.0 and start 0.2.2 (the one in store) at the command prompt... it loads... I can edit the same animation-data... I can close/restart... data is there again. Fine! Then I do --run for version 0.3.0 again... it opens the edited data that was just created/edited with 0.2.2... I try to edit... immediatly there's the command prompt message about failed write again! v0.3.0 has one or two extra settings, but the animation data format is exactly the same as it is in 0.2.2, which doesn't give any trouble. The only real difference is lot's of updated code for better graphics and the ever extending js-code is organized over more *.js files. Index.html is including 13 js-scripts now. That was only 4 (with more code in them) with version 0.2.2. It's just getting me crazy!

**Comment by OpenMindNL** [comment]
PS: Screenshot of command prompt
![cp](https://user-images.githubusercontent.com/24574832/29489774-d54f5ef4-8528-11e7-8931-2754dda01893.png)

LEDitor initializing = app.js starting
Started/Stopped animation = initializing LEDitor animation
2 times the Failed to save message = opening index.html about 30 seconds after the app has been started. Some settings for preferences etc are opened/stored.

Stopping... = Ctrl-C
Failed to save message from ???. My code doesn't save anything at closing

**Comment by OpenMindNL** [comment]
I restarted Homey 2 times, but I didn't flush/delete any data yet, and the message still appears... so here's my log;

450997270D

**Comment by OpenMindNL** [comment]
This is definitely a storage problem! I disabled and removed a few unnecessary logs from Insights, which released about 40MB of storage space ... and suddenly the `Failed to save app settings! (write)` message has disappeared!
Before I deleted the logs I still had about 95MB left according to Settings/System... enough to store more than 6000 animations but it refused to store a few bytes of settings *grin*

This also makes me wonder... why is Insights suddenly loggin so much data to storage? Now there's still more than 600MB left of which 44% seems to be statistics... what are those? Do I need them? where can I delete them?
